code/computational_geometry/src/sphere_tetrahedron_intersection/sphere_tetrahedron_intersection.cpp

```diff
@@ -0,0 +1,1237 @@
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = abs( rhs );
+		}
+	};
+
+	template< class T, class S >
+	struct OP_ARG
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = arg( rhs );
+		}
+	};
+
+	template< size_t N, class T, class S >
+	struct OP_MIN
+	{
+		inline void operator ()( LuVector< N, T >& result, const LuVector< N, T >& lhs, const LuVector< N, S >& rhs, const size_t& I )
+		{
+			result[ I ] = lhs[ I ] < rhs[ I ] ? lhs[ I ] : rhs[ I ];
+		}
+	};
+
+	template< size_t N, class T, class S >
+	struct OP_MAX
+	{
+		inline void operator ()( LuVector< N, T >& result, const LuVector< N, T >& lhs, const LuVector< N, S >& rhs, const size_t& I )
+		{
+			result[ I ] = lhs[ I ] > rhs[ I ] ? lhs[ I ] : rhs[ I ];
+		}
+	};
+
+	template< size_t N, class T, class S >
+	struct OP_DOT
+	{
+		inline void operator ()( T& result, const LuVector< N, T >& lhs, const LuVector< N, S >& rhs, const size_t& I )
+		{
+			result += lhs[ I ] * rhs[ I ];
+		}
+	};
+
+	//////////////////////////////////////////////////////////////
+
+	template< class T >
+	struct OP_OSTREAM
+	{
+		inline void operator ()( ostream& lhs, const T& rhs, const string& delimeter )
+		{
+			lhs << rhs << delimeter;
+		}
+	};
+	
+	// LuVector_Unroll.hpp
+	
+	//////////////////////////////////////////////////////////////
+	// TEMPLATE LOOP UNROLLING
+	//////////////////////////////////////////////////////////////
+
+	// VECTOR OP SCALAR
+	template< size_t I, size_t N, class T, class S, template< class, class > class OP >
+	inline void UNROLL( LuVector< N, T >& lhs, const S& rhs, OP< T, S > operation, LOOP_INDEX< I > )
+	{
+		UNROLL( lhs, rhs, operation, LOOP_INDEX< I - 1 >() );
+		operation( lhs[ I ], rhs );
+	}
+
+	template< size_t N, class T, class S, template< class, class > class OP >
+	inline void UNROLL( LuVector< N, T >& lhs, const S& rhs, OP< T, S > operation, LOOP_INDEX< 0 > )
+	{
+		operation( lhs[ 0 ], rhs );
+	}
+
+	// SCALAR OP VECTOR
+	template< size_t I, size_t N, class T, class S, template< class, class > class OP >
+	inline void UNROLL( S& lhs, const LuVector< N, T >& rhs, OP< T, S > operation, LOOP_INDEX< I > )
+	{
+		UNROLL( lhs, rhs, operation, LOOP_INDEX< I - 1 >() );
+		operation( lhs, rhs[ I ] );
+	}
+
+	template< size_t N, class T, class S, template< class, class > class OP >
+	inline void UNROLL( S& lhs, const LuVector< N, T >& rhs, OP< T, S > operation, LOOP_INDEX< 0 > )
+	{
+		operation( lhs, rhs[ 0 ] );
+	}
+
+	// VECTOR OP VECTOR
+	template< size_t I, size_t N, class T, class S, template< class, class > class OP >
+	inline void UNROLL( LuVector< N, T >& lhs, const LuVector< N, S >& rhs, OP< T, S > operation, LOOP_INDEX< I > )
+	{
+		UNROLL( lhs, rhs, operation, LOOP_INDEX< I - 1 >() );
+		operation( lhs[ I ], rhs[ I ] );
+	}
+
+	template< size_t N, class T, class S, template< class, class > class OP >
+	inline void UNROLL( LuVector< N, T >& lhs, const LuVector< N, S >& rhs, OP< T, S > operation, LOOP_INDEX< 0 > )
```

Comment:
Please don't use ALL_CAPS; stick to camel case for function names.

Reply:
OK.